File: core/src/io/github/jerukan/physics/PhysicsState.java
package io.github.jerukan.physics;

import io.github.jerukan.util.Timer;

import java.util.ArrayList;

public class PhysicsState {

    public int warp = 1;
    protected Timer timer = new Timer();
    protected ArrayList<PhysicsObject> physicsObjects = new ArrayList<PhysicsObject>();

    public void resetTimer() {
        timer.reset();
    }

    public void update() {
        for(PhysicsObject obj : physicsObjects) {
            obj.updateVectors((float) timer.getDelta());
        }
        timer.update();
    }

    public void add(PhysicsObject physicsObject) {
        physicsObjects.add(physicsObject);
    }
}

File: core/src/io/github/jerukan/physics/PhysicsState.kt
package io.github.jerukan.physics

import io.github.jerukan.util.Timer
import java.util.*

open class PhysicsState {

    var warp: Int = 1
    protected val timer: Timer = Timer()
    protected val physicsObjects: ArrayList<PhysicsObject> = ArrayList()

    fun resetTimer() {
        timer.reset()
    }

    open fun update() {
        for(obj in physicsObjects) {
            obj.updateVectors(timer.getDelta().toFloat())
        }
        timer.update()
    }

    open fun add(physicsObject: PhysicsObject) {
        physicsObjects.add(physicsObject)
    }
}